### PR TITLE
Implement missing interaction features

### PR DESCRIPTION
**Pull request by l7ssha**
# Description

THis PR tracks implementation of missing interaction features

- [x] edit original response
- [x] get original response
- [x] return message when sending followup
- [x] delete followup
- [x] delete original response
- [x] get commands
   - [x] cached
   - [x] from api
- [x] edit single command
- [x] delete command
- [x] bulk override

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] Ran `dartanalyzer --options analysis_options.yaml .`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by HarryET — DISMISSED** [review]
LGTM, might be worth adding into IHttpEndpoints at some time though?

**Comment by l7ssha** [comment]
> LGTM, might be worth adding into IHttpEndpoints at some time though?

We could have something like that in Interactions class, yes. Or just move/implement stuff there

**Comment by HarryET** [comment]
This should include #168 as it seams more applicable than #159

**Comment by l7ssha** [comment]
its ready for review. Only thing left is documentation.

File: nyxx_interactions/lib/src/Interactions.dart [hunk 1]
```diff
@@ -25,6 +22,12 @@ class Interactions {
   final _buttonHandlers = <String, ButtonInteractionHandler>{};
   final _multiselectHandlers = <String, MultiselectInteractionHandler>{};
 
+  /// Commands registered by bot
+  Iterable<SlashCommand> get commands => UnmodifiableListView(this._commands);
+
+  /// Reference to client
+  final Nyxx client;
```
Comment (HarryET) [inline nyxx_interactions/lib/src/Interactions.dart:29]:
```suggestion
  final Nyxx _client;
```
Use a private variable as the user shouldn't need to access this

**Review by HarryET — CHANGES_REQUESTED** [review]
Have a look at the comment about using _client instead of client.

**Review by HarryET — APPROVED** [review]

**Pull request merged**